frontend/src/api/common/getQueryStats.ts

```diff
@@ -12,14 +13,27 @@ interface GetQueryStatsProps {
 	setData: React.Dispatch<React.SetStateAction<WsDataEvent | undefined>>;
 }
 
+function getURL(baseURL: string, queryId: string): URL | string {
+	if (baseURL && !isEmpty(baseURL)) {
+		return `${baseURL}/api/v3/query_progress?q=${queryId}`;
+	}
+	const url = new URL(
+		`/api/v3/query_progress?q=${queryId}`,
+		window.location.href,
+	);
+	url.protocol = 'wss';
+	return url;
+}
+
 export function getQueryStats(props: GetQueryStatsProps): void {
 	const { queryId, setData } = props;
 
 	const token = getLocalStorageApi(LOCALSTORAGE.AUTH_TOKEN) || '';
-	const socket = new WebSocket(
-		`${ENVIRONMENT.wsURL}/api/v3/query_progress?q=${queryId}`,
-		token,
-	);
+
+	// https://github.com/whatwg/websockets/issues/20 reason for not using the relative URLs
```

Comment:
We can remove this comment now.

Reply:
any reason for the same ? this states the reason for not using the relative URL, the initial code is logically correct but since the ws constructor doesn't take care of it in older browsers we are using this brute approach. so would be easy for someone else reading the code to follow.

Reply:
nvm. Additional info is always good 👍🏼